test/language/expressions/async-generators/return-suspendedYield-try-finally.js

```diff
@@ -0,0 +1,50 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Caitlin Potter <caitp@igalia.com>
+esid: pending
+description: >
+  AsyncGeneratorResumeNext:
+  If completion.[[Type]] is return, and generator.[[AsyncGeneratorState]] is
+  "suspendedYield", and generator is resumed within a try-block with an
+  associated finally block, resume execution within finally.
+flags: [async]
+---*/
+
+var g = async function*() {
+  try {
+    yield 1;
+    $ERROR('Generator must be resumed in finally block.');
+  } finally {
+    yield 2;
+  }
+};
+
+var it = g();
+it.next().then(function(ret) {
+  // Initial yield
+  assert.sameValue(ret.value, 1);
+  assert.sameValue(ret.done, false);
+
+  it.return('sent-value').then(function(ret) {
+    // yield after resume
+    assert.sameValue(ret.value, 2);
+    assert.sameValue(ret.done, false);
+
+    // Completed return
+    it.next().then(function(ret) {
+      assert.sameValue(ret.value, 'sent-value');
+      assert.sameValue(ret.done, true);
+
+      // Generator is closed
+      it.next().then(function(ret) {
+        assert.sameValue(ret.value, undefined);
+        assert.sameValue(ret.done, true);
+      }).then($DONE, $DONE);
+
+    }).catch($DONE);
+    
+  }).catch($DONE);
+
+}).catch($DONE);
```

Comment:
Is there a specific reason these are nested instead of chained anyway?

```js
it.next()
.then(function(ret) {
  assert.sameValue(ret.value, 1, 'Initial yield');
  assert.sameValue(ret.done, false, 'Initial yield');

  return it.return('sent-value');
})
.then(function(ret) {
  assert.sameValue(ret.value, 2, 'Yield in finally block');
  assert.sameValue(ret.done, false, 'Yield in finally block');

  return it.next();
})
.then(function(ret) {
  assert.sameValue(ret.value, 'sent-value', 'AsyncGeneratorResolve(generator, resultValue, true)');
  assert.sameValue(ret.done, true, 'AsyncGeneratorResolve(generator, resultValue, true)');

  return it.next();
})
.then(function(ret) {
  assert.sameValue(ret.value, undefined, 'Generator is closed');
  assert.sameValue(ret.done, true, 'Generator is closed');
})
.then($DONE, $DONE);